synapse/python_dependencies.py

```diff
@@ -97,6 +97,7 @@
     "sentry": ["sentry-sdk>=0.7.2"],
     "opentracing": ["jaeger-client>=4.0.0", "opentracing>=2.2.0"],
     "jwt": ["pyjwt>=1.6.4"],
+    "lint": ["black", "flake8", "isort"],
```

Comment:
We'd like to ensure that linters are not installed when a user runs `matrix_synapse[all]`.

I've added a step to the contributor docs for installing the linters manually.